docs/docs/install/environment-variables.md

```diff
@@ -0,0 +1,173 @@
+| `PUBLIC_LOGIN_PAGE_MESSAGE`   | Public Login Page Message                    |                                             | web                                          |
+
+:::tip
+
+`TZ` is only used by the `exiftool` as a fallback in case the timezone cannot be determined from the image metadata.
+
+`exiftool` is only present in the microservices container.
+
+:::
+
+## Ports
+
+| Variable                | Description           | Default | Services         |
+| :---------------------- | :-------------------- | :-----: | :--------------- |
+| `PORT`                  | Web Port              | `3000`  | web              |
+| `SERVER_PORT`           | Server Port           | `3001`  | server           |
+| `MICROSERVICES_PORT`    | Microservices Port    | `3002`  | microservices    |
+| `MACHINE_LEARNING_PORT` | Machine Learning Port | `3003`  | machine learning |
+
+## URLs
+
+| Variable                      | Description                                              |                Default                | Services              |
+| :---------------------------- | :------------------------------------------------------- | :-----------------------------------: | :-------------------- |
+| `IMMICH_WEB_URL`              | Immich Web URL                                           |       `http://immich-web:3000`        | proxy                 |
+| `IMMICH_SERVER_URL`           | Immich Server URL                                        |      `http://immich-server:3001`      | web, proxy            |
+| `IMMICH_MACHINE_LEARNING_URL` | Immich Machine Learning URL, set `"false"` to disable ML | `http://immich-machine-learning:3003` | server, microservices |
+| `PUBLIC_IMMICH_SERVER_URL`    | Public Immich URL                                        |      `http://immich-server:3001`      | web                   |
+| `IMMICH_API_URL_EXTERNAL`     | Immich API URL External                                  |                `/api`                 | web                   |
+
+## Paths
+
+| Variable                           | Description                      |           Default            | Services              |
+| :--------------------------------- | :------------------------------- | :--------------------------: | :-------------------- |
+| `IMMICH_MEDIA_LOCATION`            | Media Location                   |          `./upload`          | server, microservices |
+| `MACHINE_LEARNING_CACHE_FOLDER`    | ML Cache Location                |           `/cache`           | machine learning      |
+| `TRANSFORMERS_CACHE`               | ML Transformers Cache Location   |           `/cache`           | machine learning      |
+| `REVERSE_GEOCODING_DUMP_DIRECTORY` | Reverse Geocoding Dump Directory | `./.reverse-geocoding-dump/` | microservices         |
+| `TYPESENSE_DATA_DIR`               | Typesense Data Directory         |           `/data`            | typesense             |
```

Comment:
I would rather move media location to general, ML cache folder and transformers cache to ML section, geocoding to it's own section, and typesense data dir to the typesense section.